spec/system/consumer/checkout/payment_spec.rb

```diff
@@ -172,6 +172,36 @@
                 expect(page).to have_content("Voucher code Not found")
               end
             end
+
+            context "with a VINE voucher", :vcr, feature: :connected_apps do
+              let!(:vine_connected_app) {
+                ConnectedApps::Vine.create(
+                  enterprise: distributor, data: { api_key: "1234568", secret: "my_secret" }
+                )
+              }
+              before do
+                allow(ENV).to receive(:fetch).and_call_original
+                allow(ENV).to receive(:fetch).with("VINE_API_URL").and_return("https://vine-staging.openfoodnetwork.org.au/api/v1")
+              end
+
+              it "adds a voucher to the order" do
+                apply_voucher "CI3922"
+
+                expect(page).to have_content "$5.00 Voucher"
+                expect(order.reload.voucher_adjustments.length).to eq(1)
+                expect(Voucher.vine.find_by(code: "CI3922", enterprise: distributor)).not_to be_nil
+              end
+
+              context "with an invalid voucher" do
+                it "show an error" do
+                  fill_in "Enter voucher code", with: "KM1891"
+                  click_button("Apply")
+
+                  expect(page).to have_content("There was an error while adding the voucher")
+                  expect(Voucher.vine.find_by(code: "non_code", enterprise: distributor)).to be_nil
```

Comment:
Well technically there could be a voucher with "KM1891" not assign to this enterprise, and or not a vine voucher, so I think it's better to test for specifically what we are looking for. With checking for `count` to be 0, It could potentially fails is another test beforehand creates an voucher that doesn't get cleaned.